lib/honeybadger/agent.rb

```diff
@@ -365,6 +365,20 @@ def stop(force = false)
       true
     end
 
+    # Sends event to events backend
+    #
+    # @param event_name [String] a string describing the event
+    # @param payload [Hash] Additional data to be sent with the event as keyword arguments
+    def event(event_type, payload={})
+      ts = DateTime.now.new_offset(0).rfc3339
+      unless payload.kind_of?(Hash)
+        logger.error("Event has non-hash payload")
+        return
+      end
```

Comment:
I think we could call Hash(payload) here, and let any errors bubble up to the user (so that they know the data type is wrong). The new Logger could do some error handling if necessary:

```suggestion
      payload = Hash(payload)
```

Reply:
What if we allowed payload to be a string?
I've logged events like this: `.event('message', 'User logged in')`
In this case, we'd take the string and wrap it to a hash:
```
{
  event_type: 'message',
  ts: '...',
  ['payload' or 'data']: 'User logged in'
}
```

Reply:
I think we want to discourage people sending plain strings.